server/updater/updater.go

```diff
@@ -0,0 +1,295 @@
+// Package updater constantly updates plugins installed to the Mattermost Server to the latest versions.
+// Check Option API to see how to configure an *Updater.
+package updater
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	dlock "github.com/ilgooz/mattermost-dlock"
+	"github.com/ilgooz/mattermost-plugin-marketplace-addon/server/marketplace"
+	"github.com/ilgooz/mattermost-plugin-marketplace-addon/server/x/xplugin"
+	"github.com/mattermost/mattermost-server/plugin"
+	"github.com/pkg/errors"
+)
+
+const (
+	// defaultUpdateInterval used as a default wait time to wait before checking for updates again.
+	defaultUpdateInterval = time.Minute * 3
+
+	// updateLockKey used to do enable a distributed lock while doing update checks and updates.
+	updateLockKey = "marketplace-addon:updater"
```

Comment:
Note that the keys are already scoped to the plugin, so prefixing with `marketplace-addon` is unnecessary.

Reply:
I created this one in case we need to issue multiple locks in this plugin. This idea seemed to me like a best practice to follow while using distributed locks in plugins. What do you think?

But, I can remove `marketplace-addon:` prefix, since as you said keys are already scoped to plugins thus having this prefix doesn't make too much sense.